src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLRouteImpl.java

```diff
@@ -96,30 +99,105 @@
 
   @Override
   public DataFetcher<Iterable<Object>> stops() {
-    return environment -> getRoutingService(environment)
-        .getPatternsForRoute()
-        .get(getSource(environment))
-        .stream()
-        .map(TripPattern::getStops)
-        .flatMap(Collection::stream)
-        .collect(Collectors.toSet());
+    return environment -> getStops(environment);
   }
 
   @Override
   public DataFetcher<Iterable<Trip>> trips() {
-    return environment -> getRoutingService(environment)
-        .getPatternsForRoute()
-        .get(getSource(environment))
-        .stream()
-        .map(TripPattern::getTrips)
-        .flatMap(Collection::stream)
-        .collect(Collectors.toSet());
+    return environment -> getTrips(environment);
   }
 
-  //TODO
   @Override
   public DataFetcher<Iterable<TransitAlert>> alerts() {
-    return environment -> List.of();
+    return environment -> {
+      TransitAlertService alertService = getRoutingService(environment).getTransitAlertService();
+      var args = new LegacyGraphQLTypes.LegacyGraphQLRouteAlertsArgs(
+              environment.getArguments());
+      Iterable<LegacyGraphQLTypes.LegacyGraphQLRouteAlertType> types = args.getLegacyGraphQLTypes();
+      if (types != null) {
+        Collection<TransitAlert> alerts = new ArrayList<>();
+        types.forEach(type -> {
+          switch (type) {
+            case Route:
+              alerts.addAll(alertService.getRouteAlerts(getSource(environment).getId()));
+              break;
+            case RouteType:
+              alerts.addAll(alertService.getRouteTypeAlerts(
+                      getSource(environment).getType(),
+                      getSource(environment).getId()
+                              .getFeedId()
+              ));
+              alerts.addAll(alertService.getRouteTypeAndAgencyAlerts(
+                      getSource(environment).getType(),
+                      getSource(environment).getAgency().getId()
+              ));
+              break;
+            case Agency:
+              alerts.addAll(
+                      alertService.getAgencyAlerts(getSource(environment).getAgency().getId()));
+              break;
+            case Trips:
+              getTrips(environment).forEach(
+                      trip -> alerts.addAll(alertService.getTripAlerts(trip.getId(), null)));
+              break;
+            case StopsOnRoute:
+              getStops(environment).forEach(stop -> {
```

Comment:
After I implemented this so it checks that the alerts for stops that are on route, I realised that maybe it would make sense to just fetch all StopAndRoute type alerts that have the route defined here as it might be useful to display alerts that warn that a stop is no longer on the route, for example.

Reply:
Changed this so that the stop doesn't need to be on the route/trip/pattern nor do the route/trip have to go through the stop in order to return these StopAndRoute/Trip alerts.